### ISSUE TITLE
Nixos Thinkpad T460s

### ISSUE THREAD

**Issue by lifeashansen** [3v1n0/libfprint#55]
I have a T460s with 138a:0090 and this doesnt work

```
services.fprintd.enable = true;
services.fprintd.tod.enable = true;
services.fprintd.tod.driver = pkgs.libfprint-2-tod1-vfs0090;
```

```
lsusb
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 3151:3020 YICHIP Wireless Device
Bus 001 Device 003: ID 8087:0a2b Intel Corp. Bluetooth wireless interface
Bus 001 Device 004: ID 04ca:7058 Lite-On Technology Corp. Integrated Camera
Bus 001 Device 005: ID 138a:0090 Validity Sensors, Inc. VFS7500 Touch Fingerprint Sensor
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
```

When I run fprintd-enroll I get this error

```
fprintd-enroll
Impossible to enroll: GDBus.Error:net.reactivated.Fprint.Error.NoSuchDevice: No devices available
```

**Comment by StephanStanisic** [comment]
Hey I have a T470s with the same nixos config, and I'm getting the same error:

```
$ sudo fprintd-list steph
No devices available
$ lsusb
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 058f:9540 Alcor Micro Corp. AU9540 Smartcard Reader
Bus 001 Device 003: ID 1199:9079 Sierra Wireless, Inc. EM7455
Bus 001 Device 004: ID 8087:0a2b Intel Corp. Bluetooth wireless interface
Bus 001 Device 005: ID 04ca:7066 Lite-On Technology Corp. Integrated Camera
Bus 001 Device 006: ID 138a:0097 Validity Sensors, Inc. 
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 002: ID 0bda:0316 Realtek Semiconductor Corp. Card Reader
$ cat .local/os/nixos.nix | grep -A 3 fprintd
  services.fprintd.enable = true;
  services.fprintd.tod.enable = true;
  services.fprintd.tod.driver = pkgs.libfprint-2-tod1-vfs0090;
  systemd.services.fprintd = {
    wantedBy = [ "multi-user.target" ];
    serviceConfig.Type = "simple";
  };

```

**Comment by lifeashansen** [comment]
> Hey I have a T470s with the same nixos config, and I'm getting the same error:
> 
> ```
> $ sudo fprintd-list steph
> No devices available
> $ lsusb
> Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
> Bus 001 Device 002: ID 058f:9540 Alcor Micro Corp. AU9540 Smartcard Reader
> Bus 001 Device 003: ID 1199:9079 Sierra Wireless, Inc. EM7455
> Bus 001 Device 004: ID 8087:0a2b Intel Corp. Bluetooth wireless interface
> Bus 001 Device 005: ID 04ca:7066 Lite-On Technology Corp. Integrated Camera
> Bus 001 Device 006: ID 138a:0097 Validity Sensors, Inc. 
> Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
> Bus 002 Device 002: ID 0bda:0316 Realtek Semiconductor Corp. Card Reader
> $ cat .local/os/nixos.nix | grep -A 3 fprintd
>   services.fprintd.enable = true;
>   services.fprintd.tod.enable = true;
>   services.fprintd.tod.driver = pkgs.libfprint-2-tod1-vfs0090;
>   systemd.services.fprintd = {
>     wantedBy = [ "multi-user.target" ];
>     serviceConfig.Type = "simple";
>   };
> ```

I still havent been able to get it to work

**Comment by ZainKergaye** [comment]
Maybe this might help you: https://github.com/jasper-clarke/t480-fingerprint-nixos

**Comment by udaemon** [comment]
hey, I don't know if your issue is nixOS related or sth OS-independant. 
I also have a T460s and finally got the fingerprint reader working following the hints in this thread: https://github.com/uunicorn/python-validity/issues/129 (I'm using arch linux.) 
maybe it helps you, too! note that you have to stick quite close to the steps described there. i. e. validity-sensors-tools as it is "shipped" may not work (I mean issuing eg. `validity-sensors-tools -t initializer`) – you'd rather take just a certain part of the python script. 
In other words: I've just reset my fp reader in the bios and got error messages from fprintd-enroll, so I tried to pair (or whatever exactly) it again. while it didn't work with recent validity-sensors-tools it worked the way described in that thread (for the 2nd time). 
good luck!
